storage_file/models/storage_file.py

```diff
@@ -149,6 +153,19 @@ def _get_url(self):
         """Retrieve file URL based on backend params."""
         return self.backend_id._get_url_for_file(self)
 
+    @api.depends("slug")
+    def _compute_internal_url(self):
+        for record in self:
+            record.internal_url = record._get_internal_url()
+
+    def _get_internal_url(self):
```

Comment:
Why do you need an other method than the compute one?

Reply:
I tend to split out the computation for single records most of the times because I hate to have too loop twice on the same recordset (by calling super) when you want to change the behavior of a computed field.
As it does not hurt, I'd like to keep it :wink: